### ISSUE TITLE
Needs to self-manage

### ISSUE THREAD

**Issue by raxod502** [raxod502/straight.el#17]
`straight.el` needs to manage itself as a package. Otherwise it remains unversioned in `versions.el`, and also `pbl` does not have its autoloads activated.

**Comment by raxod502** [comment]
I'd love to have something like this:

```
(straight-use-package '(straight :fetcher github :repo "raxod502/straight.el"))
```

Unfortunately, I don't think there's any sane way to achieve this. The basic problem is that the bootstrap code is not present in the user's Emacs, but in the `straight.el` repository. Thus either the bootstrap code would need to be put into the user's init-file, which is fragile and ugly, or everyone would need to depend on the official `straight.el` repository providing appropriate bootstrap code that could allow for other forks.

However, `straight.el` still needs to self-manage, even if I can't provide pretty, consistent syntax for it. I think that the best way to do this is to update the various caches in `bootstrap.el`. That will at least get `straight` to show up in the versions lockfile. The issue with `pbl` autoloads is best handled manually in `bootstrap.el`, I think—it's probably not a good idea to try to get `straight.el` to apply its package building mechanisms to itself.

**Comment by raxod502** [comment]
Fixed in 2cd29b85baa1fcadb4bff41831b12f42d5c26d2a.

**Comment by raxod502** [comment]
Merged.
